core/connection_db.js

```diff
@@ -235,14 +247,17 @@ Blockly.ConnectionDB.prototype.searchForClosest = function(conn, maxRadius,
   var bestConnection = null;
   var bestRadius = maxRadius;
   var temp;
+  var curDistance;
 
   // Walk forward and back on the y axis looking for the closest x,y point.
   var pointerMin = closestIndex - 1;
   while (pointerMin >= 0 && this.isInYRange_(pointerMin, conn.y, maxRadius)) {
     temp = this.connections_[pointerMin];
-    if (conn.isConnectionAllowed(temp, bestRadius)) {
+    curDistance = temp.distanceFrom(conn);
+    if (curDistance <= bestRadius &&
```

Comment:
This moves the logic for distance based connection checking outside of the connection checker. I'm trying to think of a specific example where someone would want to make a connection decision based on the distance. Perhaps using a different `bestRadius` value based on the type of connection (input / output vs next/previous).

Similar to how we have a drag check, could we have a check that also takes into account the distance (when `canConnect` is called with a distance).

Reply:
The big advantage of the current setup is that they types are clean: everything is a Connection.

I *could* make it so that we cast to RenderedConnection if `isDragging` is true, but that still means I need another param for the distance.

Essentially anything I do to `canConnect` I also need to do to `canConnectWithReason`, which is frustrating.

Reply:
Ah, what if we passed in a scope object that describes the state.
```
{
  isDragging: boolean?,
  distance: number?,
  ... 
}
```

Reply:
What's the advantage of having it be a scope object instead of just adding another arg? It doesn't change the typing/casting issue.

Reply:
Saves you from having to pass parameters around, that's all.
I think what you said about casting when there's a drag makes sense. Most checks would be done on the `Connection` type, with the exception of drag checks.